Vagrantfile

```diff
@@ -0,0 +1,170 @@
+echo 'Defaults    env_keep += "SSH_AUTH_SOCK"' > /etc/sudoers.d/root_ssh_agent
+SSHD_CONFIG="/etc/ssh/sshd_config"
+if ! grep -q "^ClientAliveInterval" $SSHD_CONFIG; then
+  echo "ClientAliveInterval 60" >> $SSHD_CONFIG
+fi
+if ! grep -q "^ClientAliveCountMax" $SSHD_CONFIG; then
+  echo "ClientAliveCountMax 10000" >> $SSHD_CONFIG
+fi
+service ssh restart
+CRON_FILE="/etc/cron.hourly/auto-stop"
+if [ ! -f $CRON_FILE ]; then
+cat<<'EOF' > $CRON_FILE
+#!/bin/bash -e
+uptime_hours=$(($(awk '{print int($1)}' /proc/uptime) / 3600))
+# stop the instance if up for more than
+if [ $uptime_hours -gt #{stop_hours} ] ; then
+  wall "Automatically stopping instance after STOP_HOURS(#{stop_hours}) of uptime..."
+  halt -p
+fi
+EOF
+chmod +x $CRON_FILE
+fi
+SCRIPT
+
+$github_clone_chef_load = <<SCRIPT
+ssh-keyscan -H github.com >> ~/.ssh/known_hosts
+cd #{home_dir}
+git clone git@github.com:chef/chef-load.git
+cd chef-load
+echo "export GITHUB_TOKEN=\"#{ENV['GITHUB_TOKEN']}\"" > .secrets
+git checkout #{latest_head_commit}
+
+EC2HOSTNAME=`curl -Ss http://169.254.169.254/latest/meta-data/public-hostname`
+#sed -i "s/fqdn = .*/fqdn = '$EC2HOSTNAME'/" dev/config.toml
+SCRIPT
+
+$enter_studio = <<SCRIPT
+cat<<EOF >/etc/profile.d/hab_studio_setup.sh
+  export GITHUB_TOKEN=#{ENV['GITHUB_TOKEN']}
+  export HAB_STUDIO_SECRET_GITHUB_TOKEN=#{ENV['GITHUB_TOKEN']}
+  export AWS_ACCESS_KEY_ID=#{ENV['AWS_ACCESS_KEY_ID']}
+  export AWS_SECRET_ACCESS_KEY=#{ENV['AWS_SECRET_ACCESS_KEY']}
+  export AZURE_CLIENT_ID=#{ENV['AZURE_CLIENT_ID']}
+  export AZURE_CLIENT_SECRET=#{ENV['AZURE_CLIENT_SECRET']}
+  export AZURE_TENANT_ID=#{ENV['AZURE_TENANT_ID']}
+
+  cd #{home_dir}/chef-load
+  source .envrc
+  if [ ! -f ~/.hab/etc/cli.toml ]; then
+    echo "Setting up HAB_ORIGIN=ubuntu"
+    mkdir -p ~/.hab/etc
+    cat<<'EOT' > ~/.hab/etc/cli.toml
+origin = "ubuntu"
+auth_token = "dev444BBB999"
+ctl_secret = "dev333ZZZ111"
+EOT
+    hab origin key generate ubuntu
+  fi
+  hab studio run 'echo "http://$(curl -Ss http://169.254.169.254/latest/meta-data/public-hostname)" > url.txt'
+  hab studio enter
+EOF
+STUDIORC="#{home_dir}/chef-load/.studiorc"
+echo 'printf "\033[0;31m>>> TWO MORE STEPS NEEDED TO RUN chef-load <<<\033[0m\n"' >> $STUDIORC
+echo 'printf "1. Run this here:\033[1;32m hab pkg install chef/chef-load \033[0m\n"' >> $STUDIORC
+echo 'printf "2. And then this:\033[1;34m hab pkg binlink chef/chef-load chef-load \033[0m\n"' >> $STUDIORC
+SCRIPT
+
+if ENV['AWS_SSH_KEY_PATH'].nil?
+  ssh_key_path = '~/.ssh/id_rsa'
+else
+  ssh_key_path = ENV['AWS_SSH_KEY_PATH']
+end
+
+Vagrant.configure('2') do |config|
+  config.vm.box = 'aws'
+  config.vm.synced_folder ".", "/vagrant", disabled: true
+
+  config.vm.provider 'aws' do |aws, override|
+    aws.access_key_id = "#{ENV['AWS_ACCESS_KEY_ID']}"
+    aws.secret_access_key = "#{ENV['AWS_SECRET_ACCESS_KEY']}"
+    aws.keypair_name = ENV['AWS_SSH_KEY_NAME']
+    #aws.instance_type = 't3.nano'       # 1CPU, .5GB RAM
```

Comment:
also these comments might not be good to check in as well.

Reply:
If you want to recommend sizes I would put them in a README but if you think that this comments will help, then leave it. This is just a nitpick.

Reply:
Ok.......